src/interface_index.rs

```diff
@@ -0,0 +1,391 @@
+//! Generates an index of WebIDL interfaces.
+//! This index is inserted where "INSERT INTERFACES HERE" appears.
+
+use std::collections::BTreeMap;
+use std::io;
+
+use html5ever::tendril::StrTendril;
+use html5ever::{local_name, namespace_url, ns, QualName};
+use markup5ever_rcdom::Handle;
+
+use crate::dom_utils::NodeHandleExt;
+
+#[derive(Default, Debug)]
+struct InterfaceInfo {
+    /// Number of times the interface definition was seen. Should be one.
```

Comment:
Let's mention that we need to keep this as a number anyways, instead of a boolean, for more graceful error handling that exposes the number of times an interface was re-defined.